versioned_hdf5/tests/test_api.py

```diff
@@ -268,6 +268,46 @@ def test_create_dataset():
         assert list(f['/_version_data/versions/version1']) == list(file['version1']) == ['test_data']
         assert list(f['/_version_data/versions/version2']) == list(file['version2']) == ['test_data', 'test_data2']
 
+
+def test_changes_dataset():
+    # Testcase similar to those on generate_data.py
```

Comment:
Is it possible to literally re-use generate_data, but only do 2 or 3 transactions? That way we always stay up-to-date with it.

Reply:
If we expose `num_transactions` as a parameter that should work. For example
```
def _test_mostly_appends_sparse(self, num_transactions):
    # ...
def test_mostly_appends_sparse_small(self):
    self._test_mostly_appends_sparse(5)
def test_mostly_appends_sparse_large(self):
    self._test_mostly_appends_sparse(2500)

```

Reply:
This is what I do in the analysis branch (exposing `num_transactions` and other parameters) to automate performance tests.

The test I've written there is a simplified version of generate_data just because I wanted to avoid having helper functions like `_modify_dss_sparse` and `modify_dss_dense` here. But the operations are basically the same.